go/teams/teams.go

```diff
@@ -83,22 +83,29 @@ func (t *Team) KBFSCryptKeys(ctx context.Context, appType keybase1.TeamApplicati
 	return t.Data.TlfCryptKeys[appType]
 }
 
-func (t *Team) SharedSecret(ctx context.Context) (ret keybase1.PerTeamKeySeed, err error) {
```

Comment:
@mlsteele can you take a look and lmk if the changes to this file are correct or there are already existing accessors that i missed?

Reply:
Given that you want the team encryption key, I like what you've done here.

But note that as far as I can tell this is the first ever use of the team encryption key. What KBFS, chat, saltpack, git, and seitan do is use a secret derived from the team secret as a symmetric key.
```
(*Team).ApplicationKey(ctx, keybase1.TeamApplication_EK)
```
That has the advantage of using reader key masks to restrict access for example from implicit admins. I'm not sure whether your system could work with a symmetric secret (I think so) or benefit from RKMs. But it's something to consider.

Derived application keys have the disadvantage that people outside the team can't encrypt for the team, but I don't think you want that for EKs.

Reply:
@oconnor663 let's discuss!

Reply:
I lean towards leaving it as is, for consistency with the userEK case, where other users need to be able to verify the PUK's signature.

Because team keys aren't used for transporting any other keys, we _could_ use a derived keypair to sign them. But at that point, it would make more sense to just MAC them with the derived secret, since they can only be verified using the secret anyway. I'd prefer not to do that, both because it would be a pretty big divergence from what we do with user keys, and also because it would be nice if we had the possibility of using teamEKs to interact with other teams in the future.

It's a shame that our standard signing format/API doesn't have a spot for signature context. I would be a huge fan of domain separating different kinds of sigs that way, but I was hesitant to add a bunch of new client and server code that wasn't strictly necessary. (We've been using JSON field names to differentiate all these signatures instead, which is only slightly sketchy.)